### PR TITLE
Add Quickfix for type mismatch that converts Sequence/Array/Iterable/Collection/(Mutable)List

### PR DESCRIPTION
**Pull request by cypressious**
Fixes #KT-19735

https://youtrack.jetbrains.com/issue/KT-19735

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertCollectionFix.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+class ConvertCollectionFix(element: KtExpression, val type: CollectionType) : KotlinQuickFixAction<KtExpression>(element) {
+    override fun getFamilyName(): String = "Convert to ${type.displayName}"
+    override fun getText() = "Convert expression to '${type.displayName}' by inserting '.${type.functionCall}'"
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val expression = element ?: return
+        val factory = KtPsiFactory(expression)
+
+        val replaced = expression.replaced(factory.createExpressionByPattern("$0.$1", expression.text, type.functionCall))
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertCollectionFix.kt]: Just `expression` is better here than `expression.text`

Comment (mglukhikh) [reply idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertCollectionFix.kt]: Also, I suppose you may need to take expression into brackets sometimes. E.g. for a call like `functionOnList(a + b)`.

Comment (mglukhikh) [reply idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertCollectionFix.kt]: Full example. A bit exotic but quite possible:
```
fun foo(a: List<String>, b: List<String>) {
    bar(a + b) // expected: bar((a + b).toTypedArray()), got erroneous bar(a + b.toTypedArray())
}
fun bar(a: Array<String>) {}
```

**Comment by cypressious** [comment]
@mglukhikh fixed

**Comment by mglukhikh** [comment]
Merged manually

**Pull request closed without merging**